pkg/helm/actions/get_repos.go

```diff
@@ -0,0 +1,271 @@
+package actions
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"helm.sh/helm/v3/pkg/repo"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"sigs.k8s.io/yaml"
+
+	"github.com/openshift/console/pkg/crypto"
+)
+
+var (
+	helmChartRepositoryGVK = schema.GroupVersionResource{
+		Group:    "helm.openshift.io",
+		Version:  "v1beta1",
+		Resource: "helmchartrepositories",
+	}
+
+	defaultRepo = unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": "helm.openshift.io/v1beta1",
+			"kind":       "HelmChartRepository",
+			"metadata": map[string]interface{}{
+				"namespace": "",
+				"name":      "redhat-helm-chart",
+			},
+			"spec": map[string]interface{}{
+				"connectionConfig": map[string]interface{}{
+					"url": "https://redhat-developer.github.io/redhat-helm-charts/index.yaml",
+				},
+			},
+		},
+	}
+)
+
+const (
+	configNamespace = "openshift-config"
+)
+
+type TLSConfigGetter interface {
+	Get() (*tls.Config, error)
+}
+
+type HelmConfigGetter struct {
+	Client     dynamic.Interface
+	CoreClient corev1.CoreV1Interface
+
+	DefaultRepoCACertificate []byte
+}
+
+func (b HelmConfigGetter) IndexFiles(helmConfigs []*HelmConfig) []*repo.IndexFile {
+	var indexFiles []*repo.IndexFile
+	for _, helmConfig := range helmConfigs {
+		indexFile, err := b.IndexFile(helmConfig)
+		if err != nil {
+			plog.Errorf("Failed to load helm IndexFile %s", err.Error())
```

Comment:
Just a heads up that this will conflict with #1862, which switches logging to klog. We might hold #1862 until after this merges.

cc @zherman0 @jhadvig

Reply:
sure